polygon/heimdall/service.go

```diff
@@ -133,19 +133,19 @@ func newSpanFetcher(client HeimdallClient, logger log.Logger) entityFetcher[*Spa
 	)
 }
 
-func (s *service) FetchLatestSpan(ctx context.Context) (*Span, error) {
+func (s *service) FetchLatestSpan(ctx context.Context) (*Span, bool, error) {
 	s.checkpointScraper.Synchronize(ctx)
 	return s.store.Spans().GetLastEntity(ctx)
 }
 
-func (s *service) FetchLatestSpans(ctx context.Context, count uint) ([]*Span, error) {
+func (s *service) FetchLatestSpans(ctx context.Context, count uint) ([]*Span, bool, error) {
```

Comment:
Not sure if the ok value is needed here. What does it indicate? That there were no spans found?

Reply:
agree with Elton, `ok` is not needed here because `FetchLatestSpans` returns a slice